server/app/mutations/grid_services/common.rb

```diff
@@ -84,6 +86,30 @@ def build_grid_service_secrets(existing_secrets)
       service_secrets
     end
 
+    def build_service_volumes()
+      service_volumes = []
+      self.volumes.each do |vol|
+        vol_spec = parse_volume(vol)
+        if vol_spec[:volume]
+          # Named volume, try to find the proper mapping for external volumes
+          # NULL stack services don't have revs, for those just use the "global" name
+          volume_name = vol_spec[:volume]
+          if self.stack.latest_rev
```

Comment:
I think we should avoid this "backlink" to `latest_rev` .. reason: we are already passing latest_rev as a hash to mutation.